src/ast/codegen_llvm.cpp

```diff
@@ -1420,15 +1420,13 @@ void CodegenLLVM::visit(Probe &probe)
       {b_.getInt8PtrTy()}, // struct pt_regs *ctx
       false);
 
-  for (auto &attach_point : *probe.attach_points) {
+  auto &attach_point = (*probe.attach_points)[0];
```

Comment:
Can you leave a comment here explaining how the parser requires probes to have at least one attach point? So next person who sees this doesn't think this could be bad memory access.

Reply:
https://github.com/iovisor/bpftrace/blob/45cbee56cc007be2030c1f9d368764d0e8a95fad/src/ast/codegen_llvm.cpp#L1423-L1424